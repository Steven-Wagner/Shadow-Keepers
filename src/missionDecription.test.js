
import React from 'react';
import {shallow} from 'enzyme';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import MissionDescription from './mission/missionDescription'
import MissionOption from './mission/missionOption/missionOption';

describe('<MissionDescription', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<MissionDescription 
                                    currentDescription={{startDescription: "This is the intro. My name is Sightseer. Would you like to join my team?", options: [{label: "Count Me In", action: Array(2)}, {label: "I'm Not Sure I'm Ready", action: Array(2)}]}}
                                    />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})

describe('<MissionOption', () => {
    it('renders button correcty', () => {
        const wrapper = shallow(<MissionOption optionData={{label: "Count Me In"}}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('function called when missionOption button pressed', ()=> {

        const mockCallBack = jest.fn()

        const wrapper = mount(<MissionOption optionData={{label: "Count Me In"}} addToPendingAction={mockCallBack}/>)
        wrapper.find('button').simulate('click');
        expect(mockCallBack).toHaveBeenCalled();
    })
})
