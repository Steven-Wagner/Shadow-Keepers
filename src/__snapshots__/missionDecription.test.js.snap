// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissionDescription renders correctly 1`] = `
<div>
  <div>
    <p>
      This is the intro. My name is Sightseer. Would you like to join my team?
    </p>
  </div>
  <div
    className="mission-options"
  >
    <MissionOptions
      key="0"
      optionData={
        Object {
          "action": Array [
            undefined,
            undefined,
          ],
          "label": "Count Me In",
        }
      }
    />
    <MissionOptions
      key="1"
      optionData={
        Object {
          "action": Array [
            undefined,
            undefined,
          ],
          "label": "I'm Not Sure I'm Ready",
        }
      }
    />
  </div>
</div>
`;

exports[`<MissionOption renders button correcty 1`] = `
<button
  onClick={[Function]}
>
  Count Me In
</button>
`;
